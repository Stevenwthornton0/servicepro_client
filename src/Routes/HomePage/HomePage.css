.infoContainer {
    width: 500px;
    line-height: 2em;
    font-size: 18px;
}

.homePage-left {
    margin-top: 2em;
}

.optionsContainer {
    line-height: 5em;
    width: fit-content;
}

.optionsContainer > div > span {
    margin-left: 10px;
    font-size: 18px;
    font-weight: 300;
}

.optionsContainer > div > .logo {
    margin-left: 0;
    font-size: 32px;
    color: #757575;
}

.homePage {
    margin-top: 25px;
    margin-right: auto;
    margin-left: auto;
    margin-bottom: 70px;
    width: 68%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.demoUser {
    background: #e0e0e0;
    width: 80%;
    padding: 10px;
    border-radius: 4px;
}

@media screen and (max-width: 1030px) {
    .homePage {
        margin-left: 75px;
    }
}

@media screen and (max-width: 800px) {
    .homePage {
        width: 68%;
        margin-left: auto;
        margin-right: auto;
    }

    .infoContainer >  h2 {
        width: fit-content;
        margin-left: auto;
        margin-right: auto;
    }

    .infoContainer > h3 {
        width: 90%;
        margin-left: auto;
        margin-right: auto;
    }

    .infoContainer > h3 {
        font-size: 16px;
    }

    .optionsContainer {
        margin-left: auto;
        margin-right: auto;
    }

    .buttonContainer {
        margin-top: 40px;
        width: 100%;
        margin-right: auto;
        margin-left: auto;
    }

    .getStarted {
        margin-top: 40px;
        margin-right: auto;
        margin-left: auto;
        width: 100%;
        border: 1px solid lightblue;
        border-radius: 4px;
        font-size: 18px;
        font-weight: 400;
        padding-top: 10px;
        padding-bottom: 10px;
        background-color: lightblue;
    }

    .dunno {
        width: 90%;
        margin-left: auto;
        margin-right: auto;
    }
}

@media screen and (max-width: 500px) {
    .demoUser {
        margin-left: auto;
        margin-right: auto;
        margin-top: 50px;
    }

    .infoContainer {
        width: 100%;
        margin-left: auto;
        margin-right: auto;
        font-size: 14px;
    }

    .infoContainer > h3 {
        font-size: 14px;
        width: 85%;
        margin-left: auto;
        margin-right: auto;
        font-weight: 400;
    }

    .infoContainer > h2 {
        font-size: 27px;
        width: fit-content;
        margin-left: auto;
        margin-right: auto;
    }

    .homePage-left {
        margin-top: 0;
    }

    .homePage {
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    .optionsContainer {
        width: fit-content;
        margin-left: auto;
        margin-right: auto;
    }

    .optionsContainer > div > span {
        margin-left: 10px;
        font-size: 14px;
        font-weight: 300;
    }
    
    .optionsContainer > div > .logo {
        margin-left: 0;
        font-size: 26px;
        color: #757575;
    }
}