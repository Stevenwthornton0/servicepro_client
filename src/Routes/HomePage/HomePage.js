import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Registration from '../../components/Registration/Registration';
import './HomePage.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ServicesContext from '../../contexts/ServicesContext';

class HomePage extends Component {

    static contextType = ServicesContext;

    handleRegistrationSuccess = user => {
        const { history } = this.props;
        history.push('/login')
    }

    componentDidMount() {
        const { open } = this.context;
        if (open === true) {
            this.context.setOpen()
        }
    }

    render() {
        return (
            <div className='homePage'>
                <div className='homePage-left'>
                    <div className='infoContainer'>
                        <h2>Welcome to ServicePRO!</h2>
                        <h3>We provide an easy connection between those seeking a service, and Service Providers.</h3>
                    </div>

                    <div className='optionsContainer'>
                        <div>
                            <FontAwesomeIcon className='logo' icon='search'/>
                            <span>Find a Service Provider for anything you need</span>
                        </div>
                        <div>
                            <FontAwesomeIcon className='logo' icon='newspaper' />
                            <span>Leave a review for our Service Providers</span>
                        </div>
                        <div>
                            <FontAwesomeIcon className='logo' icon='user-plus' />
                            <span>Register your business as a Service Provider</span>
                        </div>
                    </div>

                    <div className='dunno'>
                        <Link to='/register' className='mobile buttonContainer'>
                            <button className='getStarted'>
                                Get Started
                            </button>
                        </Link>
                    </div>
                </div>

                <div className='computer'>
                    <Registration 
                        onRegistrationSuccess={this.handleRegistrationSuccess}
                    />
                </div>
                
            </div>
        )
    }
}

export default HomePage;