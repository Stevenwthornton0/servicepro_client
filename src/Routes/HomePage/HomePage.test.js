import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import HomePage from './HomePage';

Enzyme.configure({ adapter: new Adapter() })

describe('HomePage component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<HomePage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})