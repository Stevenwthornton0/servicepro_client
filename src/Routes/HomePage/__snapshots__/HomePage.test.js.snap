// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders without crashing 1`] = `
<div
  className="homePage"
>
  <div
    className="homePage-left"
  >
    <div
      className="infoContainer"
    >
      <h2>
        Welcome to ServicePRO!
      </h2>
      <h3>
        We provide an easy connection between those seeking a service, and Service Providers.
      </h3>
    </div>
    <div
      className="optionsContainer"
    >
      <div>
        <FontAwesomeIcon
          border={false}
          className="logo"
          fixedWidth={false}
          flip={null}
          icon="search"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span>
          Find a Service Provider for anything you need
        </span>
      </div>
      <div>
        <FontAwesomeIcon
          border={false}
          className="logo"
          fixedWidth={false}
          flip={null}
          icon="newspaper"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span>
          Leave a review for our Service Providers
        </span>
      </div>
      <div>
        <FontAwesomeIcon
          border={false}
          className="logo"
          fixedWidth={false}
          flip={null}
          icon="user-plus"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        <span>
          Register your business as a Service Provider
        </span>
      </div>
    </div>
    <div
      className="dunno"
    >
      <Link
        className="mobile buttonContainer"
        to="/register"
      >
        <button
          className="getStarted"
        >
          Get Started
        </button>
      </Link>
    </div>
  </div>
  <div
    className="computer"
  >
    <Registration
      onRegistrationSuccess={[Function]}
    />
  </div>
</div>
`;
